src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -218,5 +210,108 @@ public void UpdateWorkloads(bool includePreviews = false)
+
+            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            {
+                var installer = _workloadInstaller.GetPackInstaller();
+                var packsToUpdate = GetUpdatablePacks(installer);
+                foreach (var pack in packsToUpdate)
+                {
+                    installer.DownloadToOfflineCache(pack, new DirectoryPath(_downloadToCacheOption), _includePreviews);
+                }
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private async Task<IEnumerable<string>> GetUpdatablePackageUrlsAsync(bool includePreview)
+        {
+            IEnumerable<string> packageUrls = new List<string>();
+            DirectoryPath? tempPath = null;
+
+            try
+            {
+                var manifestPackageUrls = _workloadManifestUpdater.GetManifestPackageUrls(includePreview);
+                packageUrls = packageUrls.Concat(manifestPackageUrls);
+
+                tempPath = new DirectoryPath(Path.Combine(_userHome, ".dotnet", "manifest-extraction"));
+                await UseTempManifestsToResolvePacksAsync(tempPath.Value, includePreview);
+
+                if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+                {
+                    var installer = _workloadInstaller.GetPackInstaller();
+                    var packsToUpdate = GetUpdatablePacks(installer)
+                        .Select(packInfo => _nugetPackageDownloader.GetPackageUrl(new PackageId(packInfo.ResolvedPackageId), new NuGetVersion(packInfo.Version)).Result);
+                    packageUrls = packageUrls.Concat(packsToUpdate);
+                    return packageUrls;
+                }
+                else
+                {
+                    throw new NotImplementedException();
+                }
+            }
+            finally
+            {
+                if (tempPath != null && tempPath.HasValue && Directory.Exists(tempPath.Value.Value))
+                {
+                    Directory.Delete(tempPath.Value.Value, true);
+                }
+            }
+        }
+
+        private async Task UseTempManifestsToResolvePacksAsync(DirectoryPath tempPath, bool includePreview)
+        {
+            var manifestPackagePaths = await _workloadManifestUpdater.DownloadManifestPackagesAsync(includePreview, tempPath);
+            await _workloadManifestUpdater.ExtractManifestPackagesToTempDirAsync(manifestPackagePaths, tempPath);
+            _workloadManifestProvider = new TempDirectoryWorkloadManifestProvider(tempPath.Value, _sdkVersion.ToString());
+            _workloadResolver = WorkloadResolver.Create(_workloadManifestProvider, _dotnetPath, _sdkVersion.ToString());
+        }
+
+        private IEnumerable<PackInfo> GetUpdatablePacks(IWorkloadPackInstaller installer)
```

Comment:
I guess list will use this method. And other than piping the description info, that's all